examples/foundational/28a-transcription-processor-openai.py

```diff
@@ -0,0 +1,138 @@
+            timestamp = f"[{msg.timestamp}] " if msg.timestamp else ""
+            logger.info(f"{timestamp}{msg.role}: {msg.content}")
+
+        # # Log the full transcript
+        # logger.info("Full transcript:")
+        # for msg in self.messages:
+        #     timestamp = f"[{msg.timestamp}] " if msg.timestamp else ""
+        #     logger.info(f"{timestamp}{msg.role}: {msg.content}")
+
+
+async def main():
+    async with aiohttp.ClientSession() as session:
+        (room_url, token) = await configure(session)
+
+        transport = DailyTransport(
+            room_url,
+            token,
+            "Respond bot",
+            DailyParams(
+                audio_out_enabled=True,
+                transcription_enabled=True,
+                vad_enabled=True,
+                vad_analyzer=SileroVADAnalyzer(),
+            ),
+        )
+
+        tts = CartesiaTTSService(
+            api_key=os.getenv("CARTESIA_API_KEY"),
+            voice_id="79a125e8-cd45-4c13-8a67-188112f4dd22",  # British Lady
+        )
+
+        llm = OpenAILLMService(
+            api_key=os.getenv("OPENAI_API_KEY"),
+            model="gpt-4o",
+        )
+
+        messages = [
+            {
+                "role": "system",
+                "content": "You are a helpful LLM in a WebRTC call. Your goal is to demonstrate your capabilities in a succinct way. Your output will be converted to audio so don't include special characters in your answers. Respond to what the user said in a creative, helpful, and brief way. Say hello.",
+            },
+        ]
+
+        context = OpenAILLMContext(messages)
+        context_aggregator = llm.create_context_aggregator(context)
+
+        # Create transcript processor and handler
+        transcript_processor = TranscriptProcessor()
+        transcript_handler = TranscriptHandler()
+
+        # Register event handler for transcript updates
+        @transcript_processor.event_handler("on_transcript_update")
+        async def on_transcript_update(processor, frame):
+            await transcript_handler.on_transcript_update(processor, frame)
+
+        pipeline = Pipeline(
+            [
+                transport.input(),  # Transport user input
+                context_aggregator.user(),  # User responses
+                llm,  # LLM
+                tts,  # TTS
+                transport.output(),  # Transport bot output
+                context_aggregator.assistant(),  # Assistant spoken responses
+                transcript_processor,  # Process transcripts
+            ]
+        )
+
+        task = PipelineTask(pipeline, PipelineParams(allow_interruptions=True))
+
+        @transport.event_handler("on_first_participant_joined")
+        async def on_first_participant_joined(transport, participant):
+            await transport.capture_participant_transcription(participant["id"])
+            # Kick off the conversation.
+            await task.queue_frames([LLMMessagesFrame(messages)])
```

Comment:
I think we're trying to move away from sending an `LLMMessagesFrame` to start the conversation.

I've been trying to do this instead:

```
            await task.queue_frames([context_aggregator.user().get_context_frame()])
```

The reason is that some LLM operations require that the llm service have a persistent context object. Function calling is an example. 

If we just send an `LLMMessagesFrame` the llm service creates a context to use temporarily. But then when the context_aggregator pushes the next frame, that context gets overwritten.

I've actually been wondering if we should formally deprecate starting a pipeline with an `LLMMessagesFrame` and throw a warning.

(It's fine to use an `LLMMessagesFrame` later, when there's already a context.)

Reply:
Good to know. If we want to move in this direction (and it seems we do), then we'll want to:
- document it
- (maybe) emit an error if it's used to initialize the conversation

Reply:
Fixed.